index.js

```diff
@@ -143,6 +143,11 @@ function filter(patterns, opts) {
   }
 
   patterns = utils.arrayify(patterns);
+  var len = patterns.length, i = 0;
+  while (i < len) {
+    patterns[i] = matcher(patterns[i++], opts);
```

Comment:
Should use a new array here rather than mutating `patterns` directly to prevent confusion. `patternMatchers` perhaps.